src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -114,148 +139,260 @@ def run_test(
+        :param inv_eventually_validation_sequence_set: Validation functions for invariant testing
+                                that should eventually be true, before the test ends
+        :param test_timeout_s: The timeout for the test, if any eventually_validations
+                                remain after the timeout, the test fails.
+        :param tick_duration_s: The simulation step duration
+
+        """
+
+        threading.excepthook = self.excepthook
+
+        setup(params[0])
+
+        super().run_test(
+            inv_always_validation_sequence_set,
+            inv_eventually_validation_sequence_set,
+            test_timeout_s,
+            tick_duration_s,
+        )
+
+
+class AggregateTestRunner(SimulatedTestRunner):
+
+    """
+    Runs a simulated test multiple times with different given parameters
+
+    Result of the test is determined by comparing the number of
+    passing iterations to a predetermined acceptable threshold
+    """
+
+    def __int__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def run_test(
+        self,
+        setup=(lambda: None),
+        params=[],
+        ag_always_validation_sequence_set=[[]],
+        ag_eventually_validation_sequence_set=[[]],
+        test_timeout_s=3,
+        tick_duration_s=0.0166,  # Default to 60hz
+        **kwargs,
+    ):
+        """Run an aggregate test
+
+        :param setup: Function that sets up the World state and the gamecontroller before running the test
+        :param params: List of parameters for each iteration of the test
+        :param ag_always_validation_sequence_set: Validation functions for aggregate testing
+                                that should hold on every tick
+        :param ag_eventually_validation_sequence_set: Validation functions for aggregate testing
+                                that should eventually be true, before the test ends
+        :param test_timeout_s: The timeout for the test, if any eventually_validations
+                                remain after the timeout, the test fails.
+        :param tick_duration_s: The simulation step duration
+
+        """
+
+        threading.excepthook = self.excepthook
+
+        # Runs the test once for each given parameter
+        # Catches Assertion Error thrown by failing test and increments counter
+        # Calculates overall results and prints them
+        for x in range(len(params)):
+
+            setup(params[x])
+
+            failed_tests = 0
+
+            try:
+                super().run_test(
+                    ag_always_validation_sequence_set,
+                    ag_eventually_validation_sequence_set,
+                    test_timeout_s,
+                    tick_duration_s,
+                )
+            except AssertionError:
+                failed_tests += 1
+
+        # TODO (#2856) Fix validation and results output
+
+        print(f"{failed_tests} test failed")
```

Comment:
the "failed_tests" variable is out-of-scope here, also use a logger.info statement here rather than a python print